controllers/operatorpolicy_controller.go

```diff
@@ -133,9 +146,96 @@ func (r *OperatorPolicyReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 
 	// FUTURE: more resource checks
 
+	// handle catalogsource
+	if err := r.handleCatalogSource(ctx, policy); err != nil {
+		OpLog.Error(err, "Error handling CatalogSource")
+
+		return reconcile.Result{}, err
+	}
+
 	return reconcile.Result{}, nil
 }
 
+func (r *OperatorPolicyReconciler) handleCatalogSource(
+	ctx context.Context,
+	policy *policyv1beta1.OperatorPolicy,
+) error {
+	watcher := opPolIdentifier(policy.Namespace, policy.Name)
+
+	desiredSub, err := buildSubscription(policy)
+	if err != nil {
+		return fmt.Errorf("error building Subscription: %w", err)
+	}
+
+	// Retrieve Subscription
+	foundSub, err := r.DynamicWatcher.Get(watcher, subscriptionGVK, desiredSub.Namespace, desiredSub.Name)
+	if err != nil {
+		return fmt.Errorf("error getting Subscription: %w", err)
+	}
+
+	if foundSub == nil {
+		// Subscription not found, check the CatalogSource object
+		foundCatalogSrc, err := r.DynamicWatcher.Get(watcher, catalogSrcGVK,
+			desiredSub.Spec.CatalogSourceNamespace, desiredSub.Spec.CatalogSource)
+		if err != nil {
+			return fmt.Errorf("error getting CatalogSource: %w", err)
+		}
+
+		// Conditionally emit health event using the defined status template
+		isMissing := foundCatalogSrc == nil
+
+		err = r.updateStatus(ctx, policy, catalogSourceFindCond(isMissing), catalogSourceObj(desiredSub, isMissing))
+		if err != nil {
+			return fmt.Errorf("error updating the status for CatalogSource when Subscription DNE on cluster: %w", err)
+		}
+
+		return nil
+	}
+
+	condList, found, err := unstructured.NestedSlice(foundSub.Object, "status", "conditions")
+	if !found {
+		return fmt.Errorf("field does not exist in object: %w", err)
+	}
+
+	if err != nil {
+		return fmt.Errorf("error decoding object into array type: %w", err)
+	}
+
+	// Iterate over all the conditions to determine the state of the CatalogSource
+	for _, condition := range condList {
+		condMap, ok := condition.(map[string]interface{})
+		if !ok {
+			return fmt.Errorf("conditions field is of the wrong type")
+		}
+
+		if fmt.Sprintf("%v", condMap["type"]) == catalogSrcConditionType {
```

Comment:
It took me a minute to find a reason for using `fmt.Sprintf` throughout this section... did you choose this because it saves a bunch of error-checking for types?

I think I'm warming up to the idea, I would just like a comment explaining it once somewhere.

Reply:
Forgot to mention but this was one of the implementation details that I wanted feedback on. Is this the best way to type cast `interface{}` to `string`? I chose this way because it was one that worked, but I was wondering if this was the best way to approach this.

Reply:
@JustinKuli

Reply:
I would've expected something like
```golang
condType, ok := condMap["type"].(string)
if !ok {
    // handle the situation where it isn't a string or the field wasn't in the map
}
```

Without the `, ok`, I think it could panic which we would not want.

I don't think the `fmt.Sprintf` like this will ever panic, probably gives a usefulresult even in odd situations, and at worst might just be an empty string.